js-practive/js/main.js

```diff
@@ -1 +1,196 @@
-// Place your code here, or you can split it how you what. It's up to you 😎
\ No newline at end of file
+const USER_SELECTOR = '.user';
+const EDIT_BTN_CLASS = 'editBtn';
+const DELETE_BTN_CLASS = 'deleteBtn';
+
+const usersTable = document.getElementById('users-table-list');
+const editButton = document.getElementById('create');
+const addForm = document.getElementById('users-edit')
+const saveButton = document.getElementById('saveBnt')
+const cancelButton = document.getElementById('cancel')
+
+let usersList;
+
+getUsers()
+
+editButton.addEventListener('click', toggleCreateTable)
+saveButton.addEventListener('click', saveUserInfo)
+cancelButton.addEventListener('click', cancelForm)
+usersTable.addEventListener('click', onUsersTableClick )
+
+function getUsers(){
+    fetch('/user')
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+        usersList = data
+        showUsersList(usersList);
+    });
+}
+
+function showUsersList(usersList) {
+    const html = usersList.map(generateUserHTML).join('');
+    usersTable.innerHTML = html;
+}
+
+function saveUserInfo() {
+
+    const user = getUser()
+
+    if (user.id){
+        fetch(`/user?id=${user.id}` , {
+            method: 'PUT',
+            body: JSON.stringify(user)
+        })
+        .then((res)=>{
+            if (res.ok){
+                renderUserHTM(user)
+                clearForm()
+                toggleCreateTable()
+            }
+        });
+    } else {
+        fetch('/user', {
+            method: 'POST',
+            body: JSON.stringify(user),
+        })
+        .then((res)=>{
+            if (res.ok){
+                renderUserHTM(user)
+                clearForm()
+                toggleCreateTable()
+            }
+        });
+    }
+}
+
+function renderUserHTM(user) {
+    const userValueRenderHtml = generateUserHTML(user);
+
+    usersTable.insertAdjacentHTML("beforeend", userValueRenderHtml);
+}
+
+function generateUserHTML(user) {    
+    return `
+    <tr 
+        class="user"
+        data-id="${user.id}"
+    >
+        <th>${user.fullName}</th>
+        <th>${user.profession}</th>
+        <th>${user.shortInfo}</th>
+        <th>
+            <button class="btn editBtn">Edit</button>
+            <button class="btn deleteBtn">Delete</button>
+        </th>
+    </tr>
+    `
+}
+
+function getUser() {
+    
+    const user = usersList.find(userValue => userValue.id === addForm.elements['id'].value)
+
+    return {
+        ...user,
+        fullName: addForm.elements['fullname'].value,
```

Comment:
you can write it in more compact way like `...['fullname', 'adress'].map(key => addForm.elements[key] .value)`

Instead of all manuall work.